File: 5_kyu/Perimeter_of_squares_in_a_rectangle.py
def perimeter(n: int) -> int:
    "Returns the total length of the first n+1 squares."
    sum_length = 0  # sum of one edge length per square
    lengthI = 1     # edge length of square i
    lengthI1 = 0    # edge length of square i-1
    for i in range(n+1):
        sum_length += lengthI
        lengthI, lengthI1 = lengthI+lengthI1, lengthI
    return 4*sum_length

File: 5_kyu/Perimeter_of_squares_in_a_rectangle.r
# Returns the total length of the first n+1 squares.
perimeter <- function (n) {
    sum_length <- 0    # sum of one edge length per square
    length <- c(0,1)   # edge length of square i-1 and square i
    for( i in 0:n ) {
        sum_length <- sum_length + length[2]
        length <- c(length[2], sum(length))
    }
    return( 4*sum_length )
}